### PR TITLE
Refactor/modal

### LINKED ISSUE #413: ModalDialog: show header on top
Currently, the header is centered with the content in the ModalDialog, as shown here:

https://github.com/cstenglein/raspiblitz-web/pull/412

Place the header on top & have the content centered, should look better IMO

### PR DESCRIPTION
**Pull request by escapedcat**
Fixes #413 

# First draft

Only took care of the receive modal for now to test this.  
Happy for any feedback.

## Screenshots
image

image

image

## Open points

- modal needs to be checked for all screen-sizes 
- light-mode looks weird, but it kinda looks weird on `master` too
- buttons could be optimized in general i.e.
   - go with 3 versions: primary, secondary, default
   - they all work in light- and dark-mode
   - should handle different states (i.e. disabled) and icons (i.e. spinner)
   - should be done in another PR
- switch tab-component to headlessui as well (new PR)

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -9,8 +9,8 @@ module.exports = {
         "bd-yellow-dark": "#F7CB47",
       },
     },
-    maxWidth: {
-      "4/5": "80%",
-    },
+    // maxWidth: {
+    //   "4/5": "80%",
```
Comment (cstenglein) [inline tailwind.config.js:13]: You can just delete it if we are not using it anymore :)

File: package.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
   },
   "dependencies": {
     "@bitcoin-design/bitcoin-icons-react": "^0.1.9",
+    "@headlessui/react": "^1.7.15",
```
Comment (cstenglein) [inline package.json:29]: 👍
Was playing around with it once as well

**Comment by escapedcat** [comment]
Will try to bring this to a presentable state first ;)

File: package.json (hunk 2 above)
Comment (escapedcat) [reply package.json:29]: Alby is using this and it works pretty nice

**Comment by escapedcat** [comment]
@cstenglein still just a draft, but could be applied to "all" modals. No rush, any feedback welcome.

File: src/pages/Home/ReceiveModal/ReceiveModal.tsx [hunk 3]
```diff
@@ -95,87 +94,75 @@ const ReceiveModal: FC<Props> = ({ onClose }) => {
...
     generateInvoiceHandler();
 
-  return createPortal(
-    <ModalDialog close={onClose}>
-      <div className="text-xl font-bold">
-        {showLnInvoice ? t("wallet.create_invoice_ln") : t("wallet.fund")}
-      </div>
+  const onBack = () => setAddress("");
 
+  return createPortal(
```
Comment (cstenglein) [inline src/pages/Home/ReceiveModal/ReceiveModal.tsx:107]: the transition doesn't really work with a portal. need to google how to make it work

File: src/layouts/Modal.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                <form
+                  onSubmit={(event) => {
+                    event.preventDefault();
+                    submitFunc && submitFunc();
+                  }}
+                >
+                  <div className="absolute right-0 top-0 hidden pr-4 pt-4 sm:block">
+                    <button
+                      type="button"
+                      className="rounded-md bg-white dark:bg-zinc-900 text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
```
Comment (cstenglein) [inline src/layouts/Modal.tsx:71]:
```suggestion
                      className="rounded-md bg-gray-300 dark:bg-zinc-900 text-black dark:text-white  hover:text-gray-500 hover:dark:text-gray-500 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
```

The X with white text is more clear in dark mode. Maybe should also have a box in light mode, so it is more consistent.

File: src/layouts/Modal.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                        >
+                          {title}
+                        </Dialog.Title>
+
+                        <div className="mt-2">{children}</div>
+                      </div>
+                    </div>
+                  </div>
+
+                  <div className="bg-gray-50 dark:bg-zinc-900 px-4 py-3 sm:flex justify-between sm:px-6">
```
Comment (cstenglein) [inline src/layouts/Modal.tsx:94]:
```suggestion
                  <div className="bg-gray-50 dark:bg-gray-800 px-4 py-3 sm:flex justify-between sm:px-6">
```

The contrast is too stark for that IMO, so better leave it as one.

What you could try is a horizontal line to divide it.

File: src/layouts/Modal.tsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+            <Transition.Child
+              as={Fragment}
+              enter="ease-out duration-300"
+              enterFrom="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
+              enterTo="opacity-100 translate-y-0 sm:scale-100"
+              leave="ease-in duration-200"
+              leaveFrom="opacity-100 translate-y-0 sm:scale-100"
+              leaveTo="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
+            >
+              <Dialog.Panel className="relative transform  overflow-hidden rounded-lg bg-white text-left shadow-xl transition-all sm:my-8 sm:w-full sm:max-w-lg">
```
Comment (cstenglein) [inline src/layouts/Modal.tsx:61]:
```suggestion
              <Dialog.Panel className="relative transform overflow-hidden rounded-lg bg-white text-left shadow-xl transition-all sm:my-8 sm:w-full sm:max-w-lg">
```

File: src/layouts/ModalDialog.tsx [hunk 7]
```diff
@@ -41,16 +41,17 @@ const ModalDialog: FC<Props> = ({ closeable = true, close, children }) => {
   return (
     <ModalBackground>
       <div className="xl:max-w-screen-sm flex h-screen max-h-full w-screen flex-col overflow-y-auto rounded-lg bg-white pb-4 text-center shadow-xl dark:bg-gray-800 dark:text-white md:h-auto md:w-4/5 lg:w-1/2 xl:mx-5 xl:w-2/5">
-        <div className="flex pr-2 pt-1">
+        <div className="relative">
           <button
             onClick={closeModal}
-            className={`ml-auto mt-1 flex h-7 w-7 items-end ${
+            className={`aboslute right-2 top-2 h-7 w-7 ${
```
Comment (cstenglein) [inline src/layouts/ModalDialog.tsx:47]:
```suggestion
            className={`absolute right-2 top-2 h-7 w-7 ${
```

typo.

File: src/layouts/Modal.tsx (hunk 5 above)
Comment (cstenglein) [reply src/layouts/Modal.tsx:94]: X Icon is missing on mobile

File: src/pages/Home/ReceiveModal/ReceiveAddress.tsx [hunk 8]
```diff
@@ -8,7 +8,7 @@ type Props = {
   address: string;
 };
 
-const ReceiveOnChain: FC<Props> = ({ address }) => {
+const ReceiveAddress: FC<Props> = ({ address }) => {
```
Comment (cstenglein) [inline src/pages/Home/ReceiveModal/ReceiveAddress.tsx:11]: :+1:

File: src/layouts/Modal.tsx [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+                      {submitLabel && (
+                        <button
+                          type="submit"
+                          className="inline-flex w-full bg-yellow-500 hover:bg-yellow-400 justify-center rounded-md px-3 py-2 text-sm font-semibold text-white shadow-sm sm:ml-3 sm:w-auto"
+                        >
+                          {submitLabel}
+                        </button>
+                      )}
+
+                      <button
```
Comment (cstenglein) [inline src/layouts/Modal.tsx:120]: In light mode, there is no contrast for the button (was like this before). 
=> maybe add a bg color in light mode

File: src/layouts/Modal.tsx [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+                    >
+                      <span className="sr-only">{t("navigation.cancel")}</span>
+                      <XMarkIcon className="h-6 w-6" aria-hidden="true" />
+                    </button>
+                  </div>
+
+                  <div className="bg-white px-4 pb-4 pt-5 dark:bg-gray-800 dark:text-white sm:p-6 sm:pb-4">
+                    <div className="sm:flex sm:items-start">
+                      <div className="mt-3 text-center sm:ml-4 sm:mt-0 sm:text-left">
+                        <Dialog.Title
```
Comment (cstenglein) [inline src/layouts/Modal.tsx:82]: Maybe the title should be bigger & centered in desktop as well (same with txbuttons).

**Review by cstenglein — COMMENTED** [review]
Thank you for the work on this :) :+1:

File: src/layouts/Modal.tsx (hunk 9 above)
Comment (escapedcat) [reply src/layouts/Modal.tsx:120]: Might be good to create a Button-component which takes care of this in a later PR and use this in the future

File: src/layouts/Modal.tsx (hunk 10 above)
Comment (escapedcat) [reply src/layouts/Modal.tsx:82]: - bigger is good
- I'm not so sure about "centered on desktop"
  Looking at [some known examples](https://component.gallery/components/modal/) I don't find a centered exmaple for desktop. If in the future bigger modals are used centered text will float weirdly. On smaller screens (i.e. mobile) this isn't an issue.

Comment (cstenglein) [reply src/layouts/Modal.tsx:85]: Sure I'm fine with it :)

**Comment by escapedcat** [comment]
Will rework this. Not happy with it. Should be simpler.

**Comment by escapedcat** [comment]
Learned some things. Will tackle this again after the UI has been updated or during the update.

**Pull request closed without merging**